### DISCUSSION TITLE
plugin load configuration behaviour

### DISCUSSION THREAD

**Discussion by gireeshpunathil** [eclipse-pde/eclipse.pde#994] [Q&A]
In the launch configuration plugins tab, is there any difference between `all workspace and enabled target plugins` and `plugins selected below` followed by all plugins selected in the target plus all plugins deselected from the workspace? I was debugging a p2 installation issue and I thought both are effectively same, but am seeing behavioral difference between the two options, so thought asking here.

**Comment by merks** [comment]
If I read correctly you exclude all workspace plugins with the second approach but not the first.  That’s quite different.

**Comment by gireeshpunathil** [comment]
sorry, I messed it up.

option 1: nothing in the workspace, just launch from a plain eclipse, and select `all workspace and enabled target plugins`

option 2: many plugins in the workspace (say entire p2 project). in launch, select `plugins selected below` followed by all plugins selected in the target plus all plugins deselected from the workspace

are these equal?

**Reply by laeubi** [reply]
I don't understand the use-case and why it should be equal, if you import plugins into the workspace, an equal one would be to use both times an empty workspace.

If they are in the workspace you most probably need at least disable to automatically add required plugins and bundles option and check that the one you disabled is still there but with an older version.

**Comment by gireeshpunathil** [comment]
my scenario is this: I am trying to isolate / debug https://github.com/eclipse-equinox/p2/issues/350 . 

 - if I use plain 4.30 SDK (empty workspace) and launch `p2 director` as mentioned, it fails: we get garbled source.info file.
 - if I import p2 bundles into the workspace, but exclude those in the plugin list for launch, the test passes! (all the bundles in the workspace are manually unchecked, and `Include required plugins automatically while launching` is disabled (unchecked)
 
this confused me. I fear I don't have a way to validate what got loaded and not, because I am running `p2.director` app which does not spawn a UI child, so we cannot see what were loaded and in use!

**Reply by laeubi** [reply]
I probably would more use a remote debugger (I assume you like to debug the problem and therefore need the sources).

**Comment by merks** [comment]
I think everything you launch creates a profile in the workspace of the hosting IDE.  E.g.,  `<ws>org.eclipse.pde.core/pde-junit/p2/org.eclipse.equinox.p2.engine/profileRegistry/profile.SurrogateProfileHandlerTest.profile/1701587722226.profile.gz`.  In general, look in `<ws>/.metadata/.plugins/org.eclipse.pde.core` for each launcher's details.
